src/MiniCover/Reports/OpenCoverReport.cs

```diff
@@ -0,0 +1,147 @@
+                    new XAttribute(XName.Get("hash"), Guid.NewGuid().ToString())
+                );  
+
+                var fullNameElement = new XElement(
+                    XName.Get("FullName"),
+                    new XText(assembly.Name)
+                );
+
+                var moduleNameElement = new XElement(
+                    XName.Get("ModuleName"),
+                    new XText(assembly.Name)
+                );    
+
+                Dictionary<SourceFile, int> dctSourceFileCount = new Dictionary<SourceFile, int>();
+
+                var filesElement = assembly.SourceFiles.Select(file =>
+                {                                 
+                    dctSourceFileCount.Add(file.Value, ++fileIndex);
+                    var fileElement = new XElement(
+                        XName.Get("File"),
+                        new XAttribute(XName.Get("uid"), dctSourceFileCount[file.Value]),
+                        new XAttribute(XName.Get("fullPath"), Path.Combine(result.SourcePath, file.Key))
+                    );  
+
+                    return fileElement;
+                });
+
+                var classesElement = assembly.SourceFiles.Select(file =>
+                { 
+                    var hitInstructions = file.Value.Instructions.Where(h => hits.Contains(h.Id)).ToArray();
+
+                    return file.Value.Instructions
+                        .GroupBy(instruction => new { instruction.Class })
+                        .Select(classes =>
+                    {
+                        var classElement = new XElement(
+                            XName.Get("Class")
+                        );
+
+                        var classfullNameElement = new XElement(
+                            XName.Get("FullName"),
+                            new XText(classes.Key.Class)
+                        );
+
+                        var methodsList = classes
+                            .GroupBy(instruction => new { instruction.Method, instruction.MethodFullName })
+                            .Select(method => 
+                            {
+                                var nameElement = new XElement(
+                                    XName.Get("Name"),
+                                    new XText(method.Key.MethodFullName)
+                                );
+
+                                var fileRefElement = new XElement(
+                                    XName.Get("FileRef"),
+                                    new XAttribute(XName.Get("uid"), dctSourceFileCount[file.Value])
+                                );
+
+                                int sequencePointMiniCounter = 0;
+                                bool allLinesAreVisited = true;
+
+                                var sequencePoints = method.Select(methodPoint =>
+                                {
+                                    var hitCount = hitInstructions.Count(hit => hit.Equals(methodPoint));
```

Comment:
This count will not be greater than 1, because when hitInstructions is created it doesn't keep track of how many times that instruction was hit.

Maybe hitInstructions should be a dictionary with instruction as key and number of hits as value.

I don't know if the hitCount is important for SonarQube, did you notice any strange behavior on SonarQube report?

Reply:
It works, but I copy the hitInstructions variable from your report and really doesn't understand how it works

Reply:
Now I understand the hitInstructions structure!, as you say, can be heavily upgraded if the file contains id - count, and it's not written to disk inmediatly (with a lock). Mine is 400Mb for a small program, If you don't mind, I want to close the opencover exporter and have a look to this part of the code (I'm interested in understand how it works)

Reply:
Feel free to improve any part of it.